### PR TITLE
Update cohere embeddings with binary embeddings

### PR DESCRIPTION
**Pull request by ravi03071991**
# Description

PR to update support for cohere binary embeddings.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-cohere/pyproject.toml [hunk 1]
```diff
@@ -32,7 +32,6 @@ version = "0.1.5"
 [tool.poetry.dependencies]
 python = ">=3.8.1,<4.0"
 llama-index-core = "^0.10.1"
-cohere = "^5.1.1"
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-cohere/pyproject.toml]: feels weird to have to remove the dep on cohere lol

Comment (ravi03071991) [reply llama-index-integrations/embeddings/llama-index-embeddings-cohere/pyproject.toml]: But to work with cohere binary embeddings we need cohere >= 5.1.1 version.

Comment (nerdai) [reply llama-index-integrations/embeddings/llama-index-embeddings-cohere/pyproject.toml]: I know lol. This was more of a quick fix/workaround. We have done similar in other integrations 😞 packages.

BUT, if we want to resolve it better, then I think the issue lies here, where langchain extra sets a conflicting range for `cohere`? Could be wrong. Please revert if have a better solution! 🙏

https://github.com/run-llama/llama_index/blob/ff73754c5b68e9f4e49b1d55bc70e10d18462bce/llama-index-core/poetry.lock#L2662

Comment (ravi03071991) [reply llama-index-integrations/embeddings/llama-index-embeddings-cohere/pyproject.toml]: oh yeah. Its weird that langchain uses older package version.

Comment (logan-markewich) [reply llama-index-integrations/embeddings/llama-index-embeddings-cohere/pyproject.toml]: I think I fixed this by ignoring langchain deps in the BUILD file

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**